crates/sui-indexer-alt-framework/src/lib.rs

```diff
@@ -206,9 +206,12 @@ impl<S: Store> Indexer<S> {
             self.check_first_checkpoint_consistency::<H>(&watermark)?;
         }
 
+        let initial_watermark =
```

Comment:
Shouldn't we have a similar check here to what you do for sequential pipeline where if there is already a watermark, we should be ignoring the `--first-checkpoint`? Or will that change only be applied once we have a notion of watermark tasks?